src/sdk/base/EtherspotWalletAPI.ts

```diff
@@ -32,24 +31,52 @@ export class EtherspotWalletAPI extends BaseAccountAPI {
   index: number;
   accountAddress?: string;
   predefinedAccountAddress?: string;
+  bootstrapAddress?: string;
+  multipleOwnerECDSAValidatorAddress?: string;
 
   /**
    * our account contract.
    * should support the "execFromEntryPoint" and "nonce" methods
    */
-  accountContract?: EtherspotWallet;
+  accountContract?: ModularEtherspotWallet;
 
-  factory?: EtherspotWalletFactory;
+  factory?: EtherspotWallet7579Factory;
 
   constructor(params: EtherspotWalletApiParams) {
     super(params);
     this.factoryAddress = params.factoryAddress;
     this.index = params.index ?? 0;
     this.predefinedAccountAddress = params.predefinedAccountAddress ?? null;
+    this.bootstrapAddress = Networks[params.optionsLike.chainId].contracts.bootstrap;
+    this.multipleOwnerECDSAValidatorAddress = Networks[params.optionsLike.chainId].contracts.multipleOwnerECDSAValidator;
+  }
+
+  async installModule(moduleTypeId: MODULE_TYPE, module: string, initData: string): Promise<string> {
+    const accountContract = EtherspotWallet7579__factory.connect(await this.getAccountAddress(), this.provider);
+    if (await accountContract.isModuleInstalled(moduleTypeId, module, initData)) {
+      throw new Error('the module is already installed')
+    }
+
+    const installModuleInterface = new ethers.utils.Interface(['function installModule(uint256 moduleTypeId,address module,bytes initData)']);
```

Comment:
why are we hardcoding the ABI here? Let's get it from somewhere else.

Reply:
Changes done.